services/s3/src/test/java/software/amazon/awssdk/services/s3/internal/multipart/PresignedUrlMultipartDownloaderSubscriberWiremockTest.java

```diff
@@ -235,18 +301,33 @@ private void stubPartialFailureScenario() {
                 .withBody("<Error><Code>InternalError</Code><Message>Second request failed</Message></Error>")));
     }
 
-    private void stubSinglePartResponse() {
+    private void stubConnectionReset() {
         stubFor(get(urlEqualTo(PRESIGNED_URL_PATH))
             .willReturn(aResponse()
-                .withStatus(206)
-                .withHeader("Content-Type", "application/octet-stream")
-                .withHeader("Content-Length", String.valueOf(TEST_DATA.length))
-                .withHeader("Content-Range", "bytes 0-" + (TEST_DATA.length - 1) + "/" + TEST_DATA.length)
-                .withHeader("ETag", "\"test-etag\"")
-                .withBody(TEST_DATA)));
+                .withFault(com.github.tomakehurst.wiremock.http.Fault.CONNECTION_RESET_BY_PEER)));
     }
 
     private URL createPresignedUrl() throws MalformedURLException {
         return new URL(presignedUrlBase + PRESIGNED_URL_PATH);
     }
+
+    private PresignedUrlMultipartDownloaderSubscriber createTestSubscriber() {
+        return new PresignedUrlMultipartDownloaderSubscriber(
+            s3AsyncClient,
+            PresignedUrlDownloadRequest.builder().presignedUrl(presignedUrl).build(),
+            1024L);
+    }
+
+    private Optional<SdkClientException> invokeValidateResponse(PresignedUrlMultipartDownloaderSubscriber subscriber,
```

Comment:
Hmm, I'm not sure about using reflection fro testing. We could either move validateResponse to a util class where we can test it independently or mock the response to return invalid response.